File: Upcoder/Lý thuyết đồ thị/Duyệt theo chiều sâu/DFS.cs
using System;
using System.Collections.Generic;

namespace DFS
{
    public class Program
    {
        static int nV, nE, S;
        static List<int>[] adj;
        static bool[] visited;

        static void Input()
        {
            string[] firstLine = Console.ReadLine().Split(' ');
            nV = int.Parse(firstLine[0]);
            nE = int.Parse(firstLine[1]);
            S = int.Parse(firstLine[2]);

            adj = new List<int>[nV + 1];
            for (int i = 1; i <= nV; i++)
            {
                adj[i] = new List<int>();
            }

            for (int i = 0; i < nE; i++)
            {
                string[] edge = Console.ReadLine().Split(' ');
                int x = int.Parse(edge[0]);
                int y = int.Parse(edge[1]);
                adj[x].Add(y);
                adj[y].Add(x);
            }

            visited = new bool[nV + 1];
        }

        static void DFS(int s)
        {
            Console.Write(s + " ");
            visited[s] = true;
            foreach (int v in adj[s])
            {
                if (!visited[v])
                    DFS(v);
            }
        }

        public static void Main(string[] args)
        {
            Input();
            DFS(S);
        }
    }
}

File: Upcoder/Lý thuyết đồ thị/Duyệt theo chiều sâu/DFS.py
nV, nE, S = map(int, input().split())
adj = [[] for _ in range(nV + 1)]  # Danh sách kề với kích thước nV + 1
visited = [False] * (nV + 1)       # Mảng visited với kích thước nV + 1

def input_graph():
    for _ in range(nE):
        x, y = map(int, input().split())
        adj[x].append(y)
        adj[y].append(x)

def dfs(s):
    print(s, end=" ")
    visited[s] = True
    for v in adj[s]:
        if not visited[v]:
            dfs(v)

input_graph()
dfs(S)
